File: Week 03/id_113/LeetCode455.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        res = 0

        g.sort()
        s.sort()

        g_length, s_length = len(g), len(s)
        i = j = 0

        while i < g_length and j < s_length:
            if g[i] <= s[j]:
                res += 1
                i += 1
                j += 1
            else:
                j += 1

        return res

File: Week 03/id_343/LeetCode_455.go
func findContentChildren(g []int, s []int) int {
    sort.Ints(g)
    sort.Ints(s)
    i, j, ret := 0, 0, 0
    
    for i < len(g) && j < len(s) {
        if s[j] >= g[i] {
            ret++
            i++
            j++
        } else {
            j++
        }
    }
    
    return ret
}
